data-prepper-plugins/peer-forwarder/src/main/java/com/amazon/dataprepper/plugins/processor/peerforwarder/HashRing.java

```diff
@@ -27,6 +33,8 @@ public HashRing(final PeerListProvider peerListProvider, final int numVirtualNod
         this.numVirtualNodes = numVirtualNodes;
 
         buildHashServerMap();
+
+        peerListProvider.addListener(this);
```

Comment:
QUES: I do not have a good answer for solutions. But would like to raise concern here

1. Will there be concurrency issue on reading and writing the state of the instance ?
2. Publish `this` in the constructor might add the risk of publishing the instance in an invalid state. Might not be the issue now as this is the last line but for future maintenance.

Reply:
1. I don't expect concurrency to be an issue as we don't expect the list to change at a high frequency. `buildHashServerMap()` creates a new map variable then assigns it to the class variable once it's built completely, so no caller will be reading from a "partially built" map.
2. Fair point - we're making the assumption that there's a very low chance of the object being called by the Listenable before the object has finished being created. I'll leave it for now, but could split this out of the constructor into its own method in a following PR.